Q1.cpp

```diff
@@ -0,0 +1,31 @@
+#include <vector>
+#include <string>
+
+// when both strings are empty, they are considered permutations
+bool is_permutation(const std::string& str1, const std::string& str2){
+	bool out = false;
+	
+	if(str1.size()==str2.size()){
+		
+		std::vector<bool> check(str1.size(),true); // check this initialization
+		bool found = false;
+		int j = 0;
+		out = true;
+
+		for(int i = 0; i<str1.size() && out; i++) {
+			found = false;
+			j = 0;
+			
+			while(!found && j<str2.size()) {
+				if(check[j]) {
```

Comment:
As much as I like the code, I think it will be easier to read like:
if(check[j] && str1[i]==str2[j]) {
  found = true;
  check[j] = false;
}